modules/swagger-parser/src/main/java/io/swagger/parser/SwaggerResolver.java

```diff
@@ -53,4 +63,26 @@ public Swagger resolve() {
 
         return swagger;
     }
+
+    public static class Settings {
```

Comment:
The last 3 comments is about this `addParametersToEachOperation` setting.
I understand it as it is only useful is specific use cases.

First at all, the variable could have a better name `addResourceParametersToEachOperation`.
The current behaviour (before this PR) is:
- resource parameters are parsed
- when processing operations, resource parameters are merged with operation parameters at operation level and resources parameters are emptied.

This result to a `Swagger` object with no resource parameters.  Even `path variable` are cleaned from resource parameters, which is the kind of parameter which have the most value at resource level. 
In our usage of swagger-parser, we want to keep resource parameters as is.
So the best option I saw to add this capability without changing swagger-parser behaviour is to introduce this settings.